consensus/hotstuff/pacemaker/timeout/controller.go

```diff
@@ -68,25 +70,70 @@ func (t *Controller) TimerInfo() *model.TimerInfo { return t.timerInfo }
 // New channel is created for each timer.
 // in the event the timeout is reached (specified as TimerInfo).
 // returns closed channel if no timer has been started.
-func (t *Controller) Channel() <-chan time.Time { return t.timeoutChannel }
+func (t *Controller) Channel() <-chan time.Time {
+	return t.timeoutChannel
+}
 
 // StartTimeout starts the timeout of the specified type and returns the timer info
 func (t *Controller) StartTimeout(view uint64) *model.TimerInfo {
-	if t.timer != nil { // stop old timer
+	// stop old timer and schedule stop of ticker
+	if t.timer != nil {
 		t.timer.Stop()
+		t.stopTicker()
 	}
 	duration := t.ReplicaTimeout()
 
+	// start round duration timer
 	startTime := time.Now().UTC()
 	timer := time.NewTimer(duration)
 	timerInfo := model.TimerInfo{View: view, StartTime: startTime, Duration: duration}
 	t.timer = timer
-	t.timeoutChannel = t.timer.C
+	t.timeoutChannel = make(chan time.Time, 1)
 	t.timerInfo = &timerInfo
 
+	var ctx context.Context
+	ctx, t.stopTicker = context.WithCancel(context.Background())
+
+	// when round duration is small react with faster timeout rebroadcasts
+	tickInterval := time.Duration(math.Min(float64(duration), t.cfg.MaxTimeoutObjectRebroadcastInterval))
+
+	// start a ticker to rebroadcast timeout objects on regular basis as long as we are in the same round.
+	go tickAfterTimeout(ctx, tickInterval, t.timeoutChannel, t.timer.C)
+
 	return &timerInfo
 }
 
+// tickAfterTimeout is a utility function which starts ticking after observing timeout from single-shot timer
+// The idea is that after observing single-shot event we create a ticker which will send tick events to `tickSink` channel
+// note that first timeout event is sent as well. We use context to track when to stop.
+// This approach allows to have a concurrent-safe implementation where there is no unsafe state sharing between caller and
+// ticking logic.
```

Comment:
Good one, I've made sure that goroutines are stopped in case of changing rounds but not in case it's restarted. Added a context. Here: https://github.com/onflow/flow-go/pull/3277/commits/fe039ac5104b66b45c88bf1a2fb55bd624fd75ca